vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+
+```{r}
+tr <- derive_vars_dt(
+  tr,
+  dtc = TRDTC,
+  new_vars_prefix = "A",
+  highest_imputation = "D",
+  date_imputation = "first"
+) %>%
+  derive_vars_dy(
+    reference_date = RANDDT,
+    source_vars = vars(ADT)
+  )
+```
+
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, RANDDT, TRLNKID, TRDTC, ADT, ADTF, ADY))
+```  
+
+### Re-labelling unscheduled visits
+
+If you have data collected at unscheduled visits, which are not identified as
+unique in your SDTM or you have company specific rules.  Below is an example of
+how it could be re-labelled. <add more information once I see the template code>
+
+```{r}
+tr <- mutate(
+  tr,
+  AVISIT = VISIT,
+  AVISITN = VISITNUM
+)
+```
+
+## Derive Parameters TLDn and TLPn {#parameters}
+
+As mentioned earlier, depending on your company specifications you may either
+create parameters for TLDn and TLPn in ADTR or similar dataset.
+
+Set the `AVALx/ANL01FL` variables.
+```{r}
+adtr <- tr %>%
+  mutate(
+    tmp_lesion_nr = str_sub(TRLNKID, 3),
+    PARAMCD = paste0("DIAM", tmp_lesion_nr),
+    PARAM = paste("Target Lesion", tmp_lesion_nr, "Analysis Diameter by Investigator"),
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1",
+    AVAL = TRSTRESN,
+    ANL01FL = if_else(!is.na(AVAL), "Y", NA_character_)
+  ) %>%
+  select(-tmp_lesion_nr)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr, USUBJID, TRLNKID, AVISITN),
+  display_vars = vars(USUBJID, TRLNKID, AVISIT, PARAMCD, PARAM, AVAL, ANL01FL)
+)
+```
+
+## Derive Parameters (TSD) and Analysis Flag `ANL01FL` {#tsd}
+
+Target Lesions Sum of Diameters is the sum of the diameters for lymph and
+non-lymph nodes (sum of analysis values `AVAL` from source observations TLDn &
+TLPn).  The analysis flag `ANL01FL` flags where the number of lesions assessed
+at baseline and at post-baseline match.  To assess whether the number of lesions
+expected and assessed match, you could compare lesion identifiers (`TR.TRLNKID`)
+/use the previous variables set up earlier to compare lesion expected vs lesion
+accessed  or check for target lesion response (from `RS`) =NE.
+
+```{r}
+adtr_sum <- get_summary_records(
+  adtr,
+  by_vars = vars(STUDYID, USUBJID, !!!adsl_vars, AVISIT, AVISITN),
+  analysis_var = AVAL,
+  summary_fun = function(x) sum(x, na.rm = TRUE),
+  set_values_to = vars(
+    PARAMCD = "TSD",
+    PARAM = "Target Lesions Sum of Diameters by Investigator",
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1"
+  )
+)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, PARAMCD, PARAM, AVISIT, AVAL)
+)
+```  
+
+In the below example, the analysis flag, ANL01FL= Y identifies the largest
+target lesion sum of diameter on a particular date for reporting if there were
+multiple observations on the same date and where lesion assessed at
+post-baseline match baseline.
```

Comment:
**[From Catherine Jin via Stephen Gormley]**

For cases like PD due to two lesion merges, post-baseline total number of lesions is fewer than baseline. Should we consider more complex algorithms for ANL01FL to cover such situations (and others)?  For example, looking into timepoint overall response at that visit and comparing SUMDIAM with previous visit.

Reply:
As such split and merge of target lesions are not very common, but we should provide some solution. It all depends how the SDTM mapping is done around split/merged lesions. For merging I am seen, T0X/T0Y in TULNKID/TRLNKID and TUTESTCD=MERGED. But this will reduce 1 lesions number as compared to baseline. and for split it TUGRPID will have original lesion number but extra record ( lesion), so again baseline will not match.
So, only counting number of lesions where we have split/merge if lesions will not give us correct results alone. In that case RS target lesion approach is really works well.
The other point of few we already discussed was we are doing SUM irrespective of lesions matching or not. we are creating the SUM record. To consider that SUM for any analysis will be handled using flags later on and each company can define with there approaches and data collection/mapping.

Reply:
Would be useful to understand a bit more how this alters current way